src/virtual-source-buffer.js

```diff
@@ -9,6 +9,31 @@ import work from 'webworkify';
 import transmuxWorker from './transmuxer-worker';
 import {isAudioCodec, isVideoCodec} from './codec-utils';
 
+// We create a wrapper around the SourceBuffer so that we can manage the
+// state of the `updating` property manually. We have to do this because
+// Firefox changes `updating` to false long before triggering `updateend`
+// events and that was causing strange problems in videojs-contrib-hls
+const makeWrappedSourceBuffer = function(mediaSource, mimeType) {
+  const sourceBuffer = mediaSource.addSourceBuffer(mimeType);
+  const wrapper = Object.create(null);
+
+  wrapper.updating = false;
+  wrapper.realBuffer_ = sourceBuffer;
```

Comment:
My guess is for debugging purposes, it might be nice to be able to grab a reference to the actual buffer object